Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -242,18 +403,23 @@ class ScriptMessageClientGtk final : public WebScriptMessageHandler::Client {
 
     bool supportsAsyncReply() override
     {
-        return false;
+        return m_supportsAsyncReply;
     }
-    
-    void didPostMessageWithAsyncReply(WebPageProxy&, FrameInfoData&&, API::ContentWorld&, WebCore::SerializedScriptValue&, WTF::Function<void(API::SerializedScriptValue*, const String&)>&&) override
+
+    void didPostMessageWithAsyncReply(WebPageProxy&, FrameInfoData&&, API::ContentWorld&, WebCore::SerializedScriptValue& serializedScriptValue, WTF::Function<void(API::SerializedScriptValue*, const String&)>&& completionHandler) override
     {
+        WebKitScriptMessageReply* message = webKitScriptMessageReplyCreate(serializedScriptValue, WTFMove(completionHandler));
+        gboolean resultValue;
+        g_signal_emit(m_manager, signals[SCRIPT_MESSAGE_WITH_REPLY_RECEIVED], m_handlerName, message, &resultValue);
```

Comment:
You should build the WebKitJavascriptResult and pass it to the signal

Reply:
Also the resultValue is unused, there's no reason for the signal to return a boolean